GitUI/UserControls/FileStatusList.cs

```diff
@@ -653,91 +659,119 @@ public void SelectStoredNextIndex(int defaultIndex = -1)
         {
-            Revision = revisions.FirstOrDefault();
-
-            var tuples = new List<(GitRevision revision, string summary, IReadOnlyList<GitItemStatus> statuses)>();
-            ObjectId baseRevGuid = null;
+            var tuples = new List<(GitRevision parentRev, string summary, IReadOnlyList<GitItemStatus> statuses)>();
+            var selectedRev = revisions.FirstOrDefault();
+            Revision = selectedRev;
+            if (selectedRev == null)
+            {
+                GitItemStatusesWithDescription = tuples;
+                return;
+            }
 
-            if (Revision != null)
+            if (revisions.Count == 1)
             {
-                GitRevision[] parentRevs;
-                if (revisions.Count == 1)
+                // Get the parents for the selected revision
+                tuples.AddRange(selectedRev.ParentIds?.Select(parentId =>
+                    ((GitRevision, string, IReadOnlyList<GitItemStatus>))(new GitRevision(parentId), "", null)));
+
+                // Show combined (merge conflicts) when a single merge commit is selected
+                var isMergeCommit = tuples.Count > 1;
+                if (AppSettings.ShowDiffForAllParents && isMergeCommit)
                 {
-                    // Get the parents for the selected revision
-                    parentRevs = Revision.ParentIds?.Select(item => new GitRevision(item)).ToArray();
+                    var conflicts = Module.GetCombinedDiffFileList(selectedRev.Guid);
+                    if (conflicts.Count != 0)
+                    {
+                        // Create an artificial commit
+                        var desc = _combinedDiff.Text;
+                        tuples.Add((new GitRevision(ObjectId.CombinedDiffId), desc, conflicts));
+                    }
                 }
-                else
-                {
-                    // With more than 4, only first -> selected is interesting
-                    // Limited selections: Show multi selection if more than two selected
-                    var i = revisions.Count <= 4 ? revisions.Count : 2;
-                    parentRevs = revisions.Skip(1).Take(i - 1).ToArray();
-
-                    if (AppSettings.ShowDiffForAllParents && revisions.Count == 2)
+            }
+            else
+            {
+                // With more than 4, only first -> selected is interesting
+                // Limited selections: Show multi selection if more than two selected
+                var multipleParents = revisions.Count <= 4 ? revisions.Count - 1 : 1;
+                tuples.AddRange(revisions
+                    .Skip(1)
+                    .Take(multipleParents)
+                    .Select(parentRev =>
+                        (parentRev, "",  Module.GetDiffFilesWithSubmodulesStatus(parentRev.ObjectId, selectedRev.ObjectId, selectedRev.FirstParentGuid))));
```

Comment:
micro nit:
```suggestion
                        (parentRev, "", Module.GetDiffFilesWithSubmodulesStatus(parentRev.ObjectId, selectedRev.ObjectId, selectedRev.FirstParentGuid))));
```

Reply:
> It's getting less to compare.

fixing, squashing after a few others are merged tomorrow. Too many rebases with all related changes lately, but now most are being merged, except the two big PRs
It would have been easier if I had these changes when starting to implement them...